src/DotNet.Status.Web.Tests/TelemetryControllerTests.cs

```diff
@@ -0,0 +1,81 @@
+using Xunit;
+using DotNet.Status.Web.Controllers;
+using Microsoft.Extensions.Logging;
+using Moq;
+using Microsoft.AspNetCore.Mvc;
+using Kusto.Ingest;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit.Abstractions;
+using System;
+using Microsoft.Extensions.Options;
+using Microsoft.Identity.Client;
+
+namespace DotNet.Status.Web.Tests
+{
+    public class TelemetryControllerTests
+    {
+        private readonly ITestOutputHelper _output;
+        private TelemetryController _controller;
+
+        public TelemetryControllerTests(ITestOutputHelper output)
+        {
+            _output = output;            
+        }
+
+        protected async Task SetUp(KustoOptions customOptions = null)
+        {
+            var collection = new ServiceCollection();
+            collection.AddOptions();
+            collection.AddLogging(l => {
+                l.AddProvider(new XUnitLogger(_output));
+            });
+
+            collection.Configure<KustoOptions>(options =>
+            {
+                options.KustoIngestConnectionString = customOptions != null ? customOptions.KustoIngestConnectionString : "fakekustoconnectionstring";
+                options.KustoDatabase = customOptions != null ? customOptions.KustoDatabase : "fakekustodatbase";
+            });
+
+            collection.AddScoped<TelemetryController>();
+
+            var kustoIngestionResult = new Mock<IKustoIngestionResult>().Object;
```

Comment:
If you only want a mock that does nothing, you can do `Mock.Of<IKustoIngestionResult>`. It's a bit more brief and conveys the "I don't really care what this object is" message. :-)

Reply:
That's a lot cleaner.